src/librbd/AioRequest.cc

```diff
@@ -187,7 +288,34 @@ namespace librbd {
 				 << m_object_image_extents << dendl;
 
 	  m_state = LIBRBD_AIO_WRITE_COPYUP;
-	  read_from_parent(m_object_image_extents);
+
+          if (is_copy_on_read(m_ictx, m_snap_id)) {
+	    m_ictx->copyup_list_lock.Lock();
```

Comment:
@jdurgin The ```read_from_parent``` deletion is protected by the lock acquired in ```CopyupRequest```'s destructor.  Good catch w/ the missing lock for ```append_request```.